pkg/controller/controller.go

```diff
@@ -204,6 +202,46 @@ func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetri
 	}
 }
 
+// workerWithPolling runs a worker thread that just dequeues items, processes them, and marks them done.
+// If the reconciler returns that it should poll, requeue the item after the default polling interval elapses.
+// If the reconciler returns an error, requeue the item up to maxRetries before giving up.
+// It enforces that the reconciler is never invoked concurrently with the same key.
+func workerWithPolling(queue workqueue.RateLimitingInterface, resourceType string, maxRetries int, reconciler func(key string) (bool, error)) func() {
```

Comment:
I'll modify the other reconciliation types to pass back that they shouldn't poll, and just have one function.